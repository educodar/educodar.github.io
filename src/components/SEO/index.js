import React from "react"
import PropTypes from "prop-types"
import { Helmet } from "react-helmet"
import { StaticQuery, graphql } from "gatsby"

/**
 * @todo Adopt hooks for SEO component
 * @todo Move SEO component to metax project
 * @param title
 * @param description
 * @param image
 * @param pathname
 * @param pageType
 * @param article
 */
const SEO = ({ title, description, image, pathname, pageType, article }) => (
  <StaticQuery
    query={query}
    render={({
      site: {
        siteMetadata: {
          defaultTitle,
          titleTemplate,
          defaultDescription,
          siteUrl,
          defaultImage,
        },
      },
    }) => {
      const seo = {
        title: title || defaultTitle,
        description: description || defaultDescription,
        image: `${siteUrl}${image || defaultImage}`,
        url: `${siteUrl}${pathname || "/"}`,
      }
      return (
        <Helmet
          defer={false}
          title={seo.title}
          titleTemplate={titleTemplate}
          meta={[
            {
              name: "description",
              content: seo.description,
            },
            {
              name: "image",
              content: seo.image,
            },
            {
              name: "google-site-verification",
              content: "d56gI8ef4vlvD5c60_X2KNMV7KNNlgEalpcH84eWbs8",
            },
          ]}
        />
      )
    }}
  />
)

export default SEO

const query = graphql`
  query SEO {
    site {
      siteMetadata {
        defaultTitle: title
        titleTemplate
        defaultDescription: description
        siteUrl
        defaultImage: image
      }
    }
  }
`

SEO.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  pathname: PropTypes.string,
  article: PropTypes.bool,
}

SEO.defaultProps = {
  title: null,
  description: null,
  image: null,
  pathname: null,
  article: false,
}
