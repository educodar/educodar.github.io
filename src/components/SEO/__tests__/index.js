import React from "react"
import renderer from "react-test-renderer"
import SEO from ".."

describe("SEO component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SEO />).toJSON
    expect(tree).toMatchSnapshot()
  })
})
