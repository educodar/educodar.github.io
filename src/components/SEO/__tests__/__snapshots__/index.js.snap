// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO component renders correctly 1`] = `[Function]`;
